src/App.tsx

```diff
@@ -1,14 +1,57 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [query, setQuery] = useState('');
+
+  const visibleTodos = useMemo(() => {
+    return todos.filter(todo => {
+      const matchesQuery = todo.title
+        .toLowerCase()
+        .includes(query.toLowerCase());
+
+      if (!matchesQuery) {
+        return false;
+      }
+
+      if (filter === Filter.All) {
+        return true;
+      }
+
+      return filter === Filter.Completed ? todo.completed : !todo.completed;
+    });
+  }, [filter, query, todos]);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setLoading(true);
+
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } finally {
+        setLoading(false);
+      }
```

Comment:
Consider adding a catch block to handle errors that might occur during the fetch operation. This will help in managing any issues that arise when fetching data from the server.